apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -348,6 +348,37 @@ func (p *ProvisionerAPI) Series(args params.Entities) (params.StringResults, err
 	return result, nil
 }
 
+// AvailabilityZone returns a provider-specific availability zone for each given machine entity
+func (p *ProvisionerAPI) AvailabilityZone(args params.Entities) (params.StringResults, error) {
+	result := params.StringResults{
+		Results: make([]params.StringResult, len(args.Entities)),
+	}
+	canAccess, err := p.getAuthFunc()
+	if err != nil {
+		return result, err
+	}
+	for i, entity := range args.Entities {
+		tag, err := names.ParseMachineTag(entity.Tag)
+		if err != nil {
+			result.Results[i].Error = common.ServerError(common.ErrPerm)
+			continue
+		}
+		machine, err := p.getMachine(canAccess, tag)
+		if err == nil {
+			hc, err := machine.HardwareCharacteristics()
```

Comment:
This creates a new err which won't be written into the Result below if it is != nil

Reply:
Modified to include an inner else statement 601202b#diff-25f5ee83be4cfb0b2df7af7f3462f85aR376